dd-trace-ot/src/test/groovy/datadog/opentracing/scopemanager/ScopeManagerTest.groovy

```diff
@@ -274,6 +276,33 @@ class ScopeManagerTest extends DDSpecification {
     writer == [[scope.span()]]
   }
 
+  def "Recursive continuation construction caps out"() {
+    setup:
+    def builder = tracer.buildSpan("test")
+    def scope = (ContinuableScope) builder.startActive(true)
+    scope.setAsyncPropagation(true)
+
+    when:
+    def continuation = (ContinuableScope.Continuation) scope.capture()
+    ArrayList<ContinuableScope.Continuation> conts = new ArrayList<>()
+    for (int i = 0; i < Config.DEFAULT_MAX_CONTINUATION_DEPTH; i++) {
+      conts.add(continuation)
+      def next = (ContinuableScope) continuation.activate()
+      next.setAsyncPropagation(true)
+      continuation = next.capture()
+    }
+
+    then:
+    continuation != null
```

Comment:
Would it be good to add an additional capture and assert it's null?

Reply:
Fixed, meant to go beyond and assert that it was null.